### ISSUE TITLE
Possible Bug when running getSymbols on Linux

### ISSUE THREAD

**Issue by AndrewKostandy** [joshuaulrich/quantmod#87]
Thank you for all your work on the quantmod package.

It seems there may be a bug while running `getSymbols` on Linux. I contacted the Shiny Apps Support team and we've been able to pinpoint the single line of code that works fine locally but throws an error when run on the Linux server.

```r
library(quantmod)
data <- getSymbols("USD/EUR",src="oanda",from="2006-01-01",to="2009-12-31",env=NULL)
Error in charToDate(x) :
  character string is not in a standard unambiguous format
```

The support team sent me some additional output from their server as well and suggested I contact you about this.

```r
> sessionInfo()
R version 3.2.2 (2015-08-14)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu precise (12.04.5 LTS)

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base

other attached packages:
[1] quantmod_0.4-5 TTR_0.23-1     xts_0.9-7      zoo_1.7-12

loaded via a namespace (and not attached):
[1] tools_3.2.2     grid_3.2.2      lattice_0.20-33

> getSymbols(Symbols="USD/EUR",src="oanda",from="2006/01/01",to="2009/12/31",env=NULL)
    As of 0.4-0, ‘getSymbols’ uses env=parent.frame() and
 auto.assign=TRUE by default.

 This  behavior  will be  phased out in 0.5-0  when the call  will
 default to use auto.assign=FALSE. getOption("getSymbols.env") and
 getOptions("getSymbols.auto.assign") are now checked for alternate defaults

 This message is shown once per session and may be disabled by setting
 options("getSymbols.warning4.0"=FALSE). See ?getSymbols for more details.
Error in charToDate(x) :
  character string is not in a standard unambiguous format
```

**Comment by braverock** [comment]
This works for me on linux with the following sessionInfo:

R version 3.2.3 (2015-12-10)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.4 LTS

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] quantmod_0.4-5 TTR_0.23-0     xts_0.9.874    zoo_1.7-12    

loaded via a namespace (and not attached):
[1] tools_3.2.3     grid_3.2.3      lattice_0.20-33 tcltk_3.2.3

**Comment by kippandrew** [comment]
What `download.file.method` do you do you recommend?

**Comment by joshuaulrich** [comment]
This is unlikely to be a bug in quantmod. The error is being thrown by `as.Date.character` because both attempts to parse using formats `"%Y-%m-%d"` and `"%Y/%m/%d"` fail (return `NA`).  The `traceback()` output would help determine which `as.Date` call was causing the error, but my guess is that this is an issue with R's `strptime` and/or the system `strptime`.

@kippandrew Users have been reporting issues with the default `download.file.method` on Ubuntu (maybe other *nix flavors are affected as well), so I've been recommending they use `"wget"` (see #83).

**Comment by kippandrew** [comment]
```
> traceback()
7: stop("character string is not in a standard unambiguous format")
6: charToDate(x)
5: as.Date.character(fr[, 1L], origin = "1970-01-01")
4: as.Date(fr[, 1L], origin = "1970-01-01")
3: getSymbols.oanda(Symbols = "USD/EUR", env = NULL, verbose = FALSE,
       warnings = TRUE, auto.assign = FALSE, from = "2006/01/01",
       to = "2009/12/31")
2: do.call(paste("getSymbols.", symbol.source, sep = ""), list(Symbols = current.symbols,
       env = env, verbose = verbose, warnings = warnings, auto.assign = auto.assign,
       ...))
1: getSymbols(Symbols = "USD/EUR", src = "oanda", from = "2006/01/01",
       to = "2009/12/31", env = NULL)
```

**Comment by joshuaulrich** [comment]
@kippandrew That means it doesn't like the dates in the downloaded file.  Use `debugonce(getSymbols.oanda)` or `options(error=recover)` to get a browser and look at `str(fr)` after `fr <- read.csv(tmp, skip = 4, as.is = TRUE, header = TRUE)` is run.

**Comment by braverock** [comment]
Try using ISO dates, YYYY-MM-DD, so from = "2006-01-01", to = "2009-12-31" with - rather than /  

I've had problems with dates including / many many times.

**Comment by AndrewKostandy** [comment]
I had tried - instead of / @braverock 
'library(quantmod)
data <- getSymbols("USD/EUR",src="oanda",from="2006-01-01",to="2009-12-31",env=NULL)
Error in charToDate(x) :
  character string is not in a standard unambiguous format'

**Comment by braverock** [comment]
sorry, I hadn't seen your traceback().  Josh is right, you'll need to get down one more level and look at the 'fr' variable, and the downloaded file itself.

**Comment by kippandrew** [comment]
Using download.file.method = "curl":

```R
debug: download.file(oanda.URL, destfile = tmp, quiet = !verbose)
Browse[2]>
debug: fr <- read.csv(tmp, skip = 4, as.is = TRUE, header = TRUE)
Browse[2]>
debug: fr[, 1L] <- as.Date(fr[, 1L], origin = "1970-01-01")
Browse[2]> fr
  X.hr..center.nginx..center.
1                     </body>
2                     </html>
```

**Comment by joshuaulrich** [comment]
@kippandrew Looks like the page moved from http:// to https://.  According to `download.file`, `method = "curl"` will not follow that redirection unless `extra = "-L"` or `options(download.file.extra="-L")` are specified.

If you can, I'd appreciate if you could change the `oanda.URL` from http:// to https:// inside of `getSymbols.oanda` and test if that solves the issue without passing the redirect flag to `curl`.  That might solve the issue here and in the other related issues.  Then I can make that change in quantmod and update on CRAN.

**Comment by kippandrew** [comment]
I don't really have much time to test, but I think your suspicion is probably correct.

**Comment by AndrewKostandy** [comment]
I tried it and it works now!
I used the following code:

```
options(download.file.method="curl") 
options(download.file.extra = "-L")
```
Thank you all for the support and advice. I really appreciate it!

**Comment by joshuaulrich** [comment]
@AndrewKostandy I just changed the URL from http to https in  3f56737e9ad516ffd774b7277a6125b872bb1b44.  Can you see if that solves the issue for you without changing any of the `download.file` options?

**Comment by AndrewKostandy** [comment]
I tried but I'm getting errors installing the package from Github and then it reverts back to my CRAN installed version.

The code I used:

```
library(devtools)
install_github("joshuaulrich/quantmod")
```

The Result:
```
* installing *source* package 'quantmod' ...
** libs
gfortran-4.8   -fPIC  -g -O2  -c ohlcq.f -o ohlcq.o
make: gfortran-4.8: No such file or directory
make: *** [ohlcq.o] Error 1
ERROR: compilation failed for package 'quantmod'
* removing '/Library/Frameworks/R.framework/Versions/3.2/Resources/library/quantmod'
* restoring previous '/Library/Frameworks/R.framework/Versions/3.2/Resources/library/quantmod'
Error: Command failed (1)
```

**Comment by joshuaulrich** [comment]
You need to be able to install compiled code from source.  See [Section 6.3.2 OS X in R Installation and Administration](https://cran.r-project.org/doc/manuals/r-release/R-admin.html#OS-X-packages) and [Appendix C.3](https://cran.r-project.org/doc/manuals/r-release/R-admin.html#OS-X).  I don't use a Mac, so I'm not much help beyond pointing to the relevant docs...

**Comment by joshuaulrich** [comment]
I'm closing this as fixed in 3f56737e9ad516ffd774b7277a6125b872bb1b44, since no one has reported issues with that fix.

**Comment by Abioy** [comment]
another problem: oanda used cloudflare CDN which ask for completing a CAPTCHA, download.file in getSymbols got a http status code 403 here :( @joshuaulrich 

R 3.2.2 with quantmod 0.4-6 under OS X 10.11.6
